internal/mode/static/state/graph/certificate_bundle.go

```diff
@@ -0,0 +1,75 @@
+package graph
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	v1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+// CAKey certificate key for optional root certificate authority.
+const CAKey = "ca.crt"
```

Comment:
Would like to point out here this is 100% hardcoded and this will need to be publically documented (no idea where or how that should be adjusted) just wanted to call that out here.

If a root certificate is named anything else inside the secret (similar to the configmap implementation) it will be ignored but with a secret this time it will be 100% ignored and offer no user output (this is intentional as ssl key pairs may not have a root ca attached to them).

Reply:
Thanks, we do have a ticket [here](https://github.com/nginx/nginx-gateway-fabric/issues/3121) for updating the documentation, we'll be sure to include that.

I'm assuming this is the field that cert-manager will always write the ca to?

Reply:
yes when cert-manager creates the secret it uses the following format and follows the serviceaccount key format as well of `ca.crt`

https://cert-manager.io/docs/usage/certificate/#additional-certificate-output-formats